### DISCUSSION TITLE
Possibility to set port-channel id for uplink of l2leaf

### DISCUSSION THREAD

**Discussion by kmueller68** [aristanetworks/avd#2511] [Q&A] [answered]
Hello.

is there a way with eos_designs to assign a fixed port-channel id for the uplink port-channel on l2leaf single or mlag switches connected to l3leaf switches?

for example:
single l2leaf switch:
Interface port-channel 1
connected to l3leaf mlag pair.

dual/mlag l2leaf switch pair:
Interface port-channel 1
connected to l3leaf mlag pair.

thank you very much,
Karl

**Comment by ClausHolbechArista** [comment, marked as answer]
This is not possible today, without a lot of custom structured configuration. We auto allocate the port-channel id based on the first interface. This is to ensure no overlap when auto creating port-channels for endpoints etc. I would strongly suggest to stick with the automatic numbering scheme. If this cannot work for your case, AVD must be enhanced to allow this to be user defined.

**Reply by gusmb** [reply]
A pre-defined offset value to ensure non-overlapping assignments between underlay and endpoints could solve this. The above is a common seen requirement and I've seen it as well. The offset value could be tuneable and dependant on the topology size in terms of number of underlay uplinks. Do the first X port channel IDs are only assignable automatically to underlay, and X+1 onwards to endpoints. User defined option 8s also handy, even together with dynamic mappings, however it would force to use build time validations to ensure that there are no collisions...

**Reply by ClausHolbechArista** [reply]
I think the offset would make it much harder to operate the network afterwards. Endpoints would be offset with ex. 2 or 3 from their interface number. So Po15 would be eth12. Not as intuitive. Can I ask _why_ it is important to set the uplink port-channel to something else? IMO in an automated world we have to rely on automatic assignments (algorithmic or from pools).

We could expose a variable to allow control, but it would only take effect on the child node, and it would come with a warning, that you have to take care of avoiding collisions yourself.
